src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -422,14 +476,214 @@ export class DeployCommand implements Command {
+            title: `Parameter: ${paramName}`,
+            value: deploymentParameter.value,
+            placeHolder: `Please enter value for parameter "${paramName}"`,
+          };
+          paramValue = await vscode.window.showInputBox(options);
+        }
+      }
+
+      if (paramValue != undefined) {
+        const updatedDeploymentParameter: BicepUpdatedDeploymentParameter = {
+          name: paramName,
+          value: paramValue,
+          isSecure: deploymentParameter.isSecure,
+          parameterType: deploymentParameter.parameterType,
+        };
+        updatedDeploymentParameters.push(updatedDeploymentParameter);
+      }
+    }
+
+    return [
+      bicepDeploymentParametersResponse.parametersFileName,
+      updatedDeploymentParameters,
+    ];
+  }
+
+  private async updateParametersFile(
+    _context: IActionContext,
+    documentPath: string,
+    parametersFileExists: boolean,
+    parametersFileName: string
+  ) {
+    let placeholder: string;
+    let parametersFileUpdateOptionString = "None";
+    let parametersFileUpdateOption: ParametersFileUpdateOption;
+    if (parametersFileExists) {
+      parametersFileUpdateOptionString = "Update";
+      parametersFileUpdateOption = ParametersFileUpdateOption.Update;
+      placeholder = `Update ${parametersFileName} with values used in this deployment?`;
+    } else {
+      const folderContainingSourceFile = path.dirname(documentPath);
+      const parametersFilePath = path.join(
+        folderContainingSourceFile,
+        parametersFileName
+      );
+      if (fse.existsSync(parametersFilePath)) {
+        parametersFileUpdateOptionString = "Overwrite";
+        parametersFileUpdateOption = ParametersFileUpdateOption.Overwrite;
+        placeholder = `File ${parametersFileName} already exists. Do you want to overwrite it?`;
+      } else {
+        parametersFileUpdateOptionString = "Create";
+        parametersFileUpdateOption = ParametersFileUpdateOption.Create;
+        placeholder = `Create parameters file from values used in this deployment?`;
+      }
+    }
+
+    const result: IAzureQuickPickItem = await _context.ui.showQuickPick(
+      this._yesNoQuickPickItems,
+      {
+        canPickMany: false,
+        placeHolder: placeholder,
+        suppressPersistence: true,
+      }
+    );
+
+    _context.telemetry.properties.parametersFileUpdateOption =
+      parametersFileUpdateOptionString;
+    if (result == this._yes) {
+      return parametersFileUpdateOption;
+    } else {
+      return ParametersFileUpdateOption.None;
+    }
+  }
+
+  private async selectValueForParameterOfTypeExpression(
+    _context: IActionContext,
+    paramName: string,
+    paramValue: string | undefined
+  ) {
+    const quickPickItems: IAzureQuickPickItem[] = [];
+    if (paramValue) {
+      const useExpressionValue: IAzureQuickPickItem = {
+        label: localize(
+          "useExpressionValue",
+          `Use value of expression "${paramValue}"`
```

Comment:
Better wording: "Use default value (`resourceGroup().location`)

Reply:
I like saying "Use value of `${paramValue}`" as a minor change

Reply:
Updated here:
https://github.com/Azure/bicep/commit/fbec291ad55d0ba35fab11577ec5e7d1d4da760b

![image](https://user-images.githubusercontent.com/30270536/172242914-791a5181-dedc-4f87-a8a1-137746bb72a7.png)

Reply:
I like the double quotes. You're good!

Reply:
> I like saying "Use value of `${paramValue}`" as a minor change. I think saying Use default value is confusing because we don't give this option for "param thing string = "hello"" even though that has a default as well

Yes, exactly!  Thus my complaint here: https://github.com/Azure/bicep/pull/7030#discussion_r890573620

I guess we'll wait for feedback again.